Update documentation init_process_group optional backend

H-Huang opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #94543

Update documentation for `init_process_group()` to mention the `backend` argument is optional.

H-Huang committed:
Update documentation init_process_group optional backend

[ghstack-poisoned]

H-Huang committed:
Update on "Update documentation init_process_group optional backend"




[ghstack-poisoned]

kwen2501 commented on test/distributed/test_c10d_common.py:
```diff
@@ -1532,6 +1532,18 @@ def tearDown(self):
+                    world_size=self.world_size,
+                )
+                dist.destroy_process_group()
+
```
Would you like to add two all_reduce calls (one with cpu, the other with gpu) to make sure it is actually working? :)
If we already have such test, please ignore me :)
Also, it can be in a separate PR if we don't have it yet.

H-Huang replied:
Sounds good, it should work since we support gloo + nccl subgroups which would follow similar logic as the new backends. I will definitely refactor the existing tests for dispatchable collectives to include this though

kwen2501 reviewed (approved):
Looks good! Thanks!

H-Huang commented:
@pytorchbot merge

H-Huang committed:
Update on "Update documentation init_process_group optional backend"


Update documentation for `init_process_group()` to mention the `backend` argument is optional.


[ghstack-poisoned]

H-Huang commented:
@pytorchbot merge

H-Huang commented:
@pytorchbot merge
